snapshot-replicator/functions/remove_snapshots.py

```diff
@@ -1,31 +1,34 @@
 import boto3  
 import datetime  
 import os
+import botocore
+
 
 source_region = os.environ['SOURCE_REGION']
 target_region = os.environ['TARGET_REGION']
-duration = os.environ['RETENTION']
-iam = boto3.client('iam')  
 instances = os.environ['DB_INSTANCES']
 
 print('Loading function')
 
 def lambda_handler(event, context):
-    source = boto3.client('rds', region_name=source_region)
-    for instance in instances.split(','):
-        for snapshot in target.describe_db_snapshots(DBInstanceIdentifier=instance, MaxRecords=50)['DBSnapshots']:
-            create_ts = snapshot['SnapshotCreateTime'].replace(tzinfo=None)
-            if create_ts < datetime.datetime.now() - datetime.timedelta(days=int(duration)):
-                print "Deleting snapshot id:", snapshot['DBSnapshotIdentifier']
-                target.delete_db_snapshot(
-                    DBSnapshotIdentifier=snapshot['DBSnapshotIdentifier']
-                )
-    target = boto3.client('rds', region_name=target_region)
-    for instance in instances.split(','):
-        for snapshot in target.describe_db_snapshots(DBInstanceIdentifier=instance, MaxRecords=50)['DBSnapshots']:
-            create_ts = snapshot['SnapshotCreateTime'].replace(tzinfo=None)
-            if create_ts < datetime.datetime.now() - datetime.timedelta(days=int(duration)):
-                print "Deleting snapshot id:", snapshot['DBSnapshotIdentifier']
-                target.delete_db_snapshot(
-                    DBSnapshotIdentifier=snapshot['DBSnapshotIdentifier']
-                )
+    def deleteSnapshots(region):
+        for instance in instances.split(','):
+            rds = boto3.client('rds', region_name=region)
+            paginator = rds.get_paginator('describe_db_snapshots')
+            page_iterator = paginator.paginate(DBInstanceIdentifier=instance)
+            snapshots = []
+            for page in page_iterator:
+                 snapshots.extend(page['DBSnapshots'])
+                 for snapshot in snapshots:
```

Comment:
I think indentation is wrong here, the second `for` (`for snapshot in snapshots`) should go outside the page iterator `for`